### PR TITLE
Allowing let identifier script mode

### PR DESCRIPTION
**Pull request by mmckeaveney**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | #6719 
| Patch: Bug Fix?          | Y
| Major: Breaking Change?  | Y
| Minor: New Feature?      | N
| Tests Added + Pass?      | Y
| Documentation PR         | N
| Any Dependency Changes?  | N
| License                  | MIT

Allowing the `let` keyword to be used as an identifier in non-strict (script) mode. This addresses the second checklist issue mentioned in #6719.

### CODE REVIEW

**Comment by loganfsmyth** [comment]
Sorry this never got looked at. Would you be up for rebasing?

**Comment by mmckeaveney** [comment]
@loganfsmyth sure, just rebase from master?

**Comment by nicolo-ribaudo** [comment]
Yes :slightly_smiling_face:

File: packages/babylon/test/fixtures/core/uncategorised/558/actual.js [hunk 1]
```diff
@@ -0,0 +1 @@
+const let = 4;
```
Comment (nicolo-ribaudo) [inline packages/babylon/test/fixtures/core/uncategorised/558/actual.js]: This isn't valid (https://tc39.github.io/ecma262/#sec-let-and-const-declarations-static-semantics-early-errors)

Comment (mmckeaveney) [reply packages/babylon/test/fixtures/core/uncategorised/558/actual.js]: cool, I'll just get rid of this test

**Comment by mmckeaveney** [comment]
@nicolo-ribaudo comments addressed. Feel free to have another quick glance when you get some time 😄

**Comment by nicolo-ribaudo** [comment]
This has been fixed by 178f2d79497e2708fb28bf58f0c7aab9e0fdef12, which takes a different approach.

**Pull request closed without merging**